src/content/en/updates/2018/03/clipboardapi.md

```diff
@@ -0,0 +1,270 @@
+
+    navigator.clipboard.writeText('Text to be copied')
+      .then(() => {
+        console.log('Text copied to clipboard');
+      })
+      .catch(err => {
+        // This can happen if the user denies clipboard permissions:
+        console.error('Could not copy text: ', err);
+      });
+
+Similarly, we can write this as an [async function], then await the return of
+`writeText()`:
+
+    async function copyPageUrl() {
+      try {
+        await navigator.clipboard.writeText(location.href);
+        console.log('Page URL copied to clipboard');
+      } catch (err) {
+        console.error('Failed to copy: ', err);
+      }
+    }
+
+
+## Paste: Reading Text from the Clipboard
+
+Much like copy, text can be read from the clipboard by calling `readText()` and
+waiting for the returned Promise to resolve with the text:
+
+    navigator.clipboard.readText()
+      .then(text => {
+        console.log('Pasted content: ', text);
+      })
+      .catch(err => {
+        console.error('Failed to read clipboard contents: ', err);
+      });
+
+For consistency, here's the equivalent async function:
+
+    async function getClipboardContents() {
+      try {
+        const text = await navigator.clipboard.readText();
+        console.log('Pasted content: ', text);
+      } catch (err) {
+        console.error('Failed to read clipboard contents: ', err);
+      }
+    }
+
+
+### Handling Paste Events
+
+There are plans to introduce a new event for detecting clipboard changes, but
+for now it's best to use the "paste" event. It works nicely with the new
+asynchronous methods for reading clipboard text:
+
+    document.addEventListener('paste', event => {
+      event.preventDefault();
+      navigator.clipboard.getText().then(text => {
+        console.log('Pasted text: ', text);
+      });
+    });
+
+
+## Security and Permissions
+
+Clipboard access has always presented a security concern for browsers. Without
+proper permissions in place, a page could silently copy all manner of malicious
+content to a user's clipboard that would produce catastrophic results when
+pasted.  Imagine a web page that silently copies `rm -rf /` or a [decompression
+bomb image] to your clipboard.
+
+Giving web pages unfettered read access to the clipboard is even more
+troublesome. Users routinely copy sensitive information like passwords and
+personal details to the clipboard, which could then be read by any page without
+them ever knowing.
+
+As with many new APIs, [`navigator.clipboard`] is only supported for pages
+served over HTTPS. To help prevent abuse, clipboard access is only allowed when
+a page is the active tab. Pages in active tabs can write to the clipboard
+without requesting permission, but reading from the clipboard always requires
+permission.
+
+To make things easier, two new permissions for copy & paste have been added to
+the [Permissions API]. The Permissions API is available in Chrome under an
```

Comment:
Permissions is on by default since 43: https://www.chromestatus.com/feature/6376494003650560

Reply:
That was my thought as well, but the API throws when the flag is turned off. Maybe it's a bug?